### PR TITLE
rise of robots

### PR DESCRIPTION
**Pull request by chrisdralla**

### CODE REVIEW

**Comment by tgaff** [comment]
@chrisdralla Could you please move your homework into a hw_6 directory?

File: ChrisDralla-chrisdralla/hw_6/rise_of_robots.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    origins = ["San Francsico", "Los Angeles", "Las Vegas", "Chicago", "New York"]
+
+    number_of_robots.to_i.times do |i|
+      name = names.sample
+      type = types.sample
+      origin = origins.sample
+
+      robot = Robot.new(name, type, origin)
+      binding.pry
+      puts "I've created #{robot}. It's name is #{name}. It's type is #{type}. It's origin is #{origin}."
```
Comment (chylaudes) [inline ChrisDralla-chrisdralla/hw_6/rise_of_robots.rb]: To test if each robot has laser_fighting_skills you can put this:
```

      puts "I've created #{robot}. It's name is #{name}. It's type is #{type}. It's origin is #{origin}. Has #{robot.laser_fighting_skills}"
```

File: ChrisDralla-chrisdralla/hw_6/steps_to_class.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    @weight = weight
+    #hometown will not be able to change because only attr_reader is used
+    @hometown = "Los Altos"
+  end
+
+  #intance method: defines behavior that only an instance of a class has access to
+  def scream
+    "AHHHH!!!!!"
+  end
+
```
Comment (chylaudes) [inline ChrisDralla-chrisdralla/hw_6/steps_to_class.rb]: LOL

File: ChrisDralla-chrisdralla/hw_6/steps_to_class.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  #Class method: method that provides that provides a behavior for the overall Class; can be accessed or called outside of the Class
+  #Class method usually begins with self.<name>
+  #Class method is usually used when you need to have access to a behavior that does not involve one instance of the Class
+
+end
+
+chris = Person.new(30, 72, 180)
+binding.pry
```
Comment (chylaudes) [inline ChrisDralla-chrisdralla/hw_6/steps_to_class.rb]: Great job going above and beyond!  I hope you're using binding.pry to make instances of your class Person.

File: ChrisDralla-chrisdralla/hw_6/thermostat.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+class Temperature
+  #definte class attributes
+  attr_accessor :current_temp, :target_temp
+
+  #create instance of Temperature
+  def initialize(current_temp, target_temp)
+    @current_temp = current_temp
+    @target_temp = target_temp
+  end
+
```
Comment (chylaudes) [inline ChrisDralla-chrisdralla/hw_6/thermostat.rb]: Try creating a class method that takes an array of temperatures and for each temp invoke the set_temp method on it 
```
 def self.get_temp(temperatures, target_temp)
#grabbing the temperatures array and using set_temp on each temp
  end
```

File: ChrisDralla-chrisdralla/hw_6/thermostat.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+end
+
+#create an array of temperatures
+temperatures = [20,40,65,66,67,68,69,70,71,72,73,74,75,76,100,101]
+
+#set a target temperature
+target_temp = 75
+
+#create an instance of each temperature
+test_temp = Temperature.new(temperatures.sample, target_temp)
```
Comment (chylaudes) [inline ChrisDralla-chrisdralla/hw_6/thermostat.rb]: using .sample is pretty creative!

**Comment by chylaudes** [comment]
Great job! I can tell you're very familiar with instance methods. Try experimenting with class methods, by using ` def self.(whatever method you want).  end `  Feel free to fill in any corrections.

**Pull request merged**